pkg/sql/pg_catalog.go

```diff
@@ -1166,14 +1169,48 @@ func makeAllRelationsVirtualTableWithDescriptorIDIndex(
 		sc catalog.SchemaDescriptor, table catalog.TableDescriptor, lookup simpleSchemaResolver,
 		addRow func(...tree.Datum) error,
 	) error,
+	populateFromType func(h oidHasher, nspOid tree.Datum, owner tree.Datum, typ *types.T, addRow func(...tree.Datum) error,
+	) error,
 ) virtualSchemaTable {
 	populateAll := func(ctx context.Context, p *planner, dbContext catalog.DatabaseDescriptor, addRow func(...tree.Datum) error) error {
 		h := makeOidHasher()
-		return forEachTableDescWithTableLookup(ctx, p, dbContext, virtualOpts,
+		if err := forEachTableDescWithTableLookup(
+			ctx,
+			p,
+			dbContext,
+			virtualOpts,
 			func(db catalog.DatabaseDescriptor, sc catalog.SchemaDescriptor, table catalog.TableDescriptor, lookup tableLookupFn) error {
 				return populateFromTable(ctx, p, h, db, sc, table, lookup, addRow)
-			})
+			},
+		); err != nil {
+			return err
+		}
+
+		// Generate rows for user defined types.
```

Comment:
can you clarify in the comment if this includes composite types as well as enum types?